src/workerd/api/http.h

```diff
@@ -1096,6 +1159,9 @@ class FetchEvent: public ExtendableEvent {
 
   void visitForGc(jsg::GcVisitor& visitor) {
     visitor.visit(request);
+    KJ_IF_SOME(respondWithCalled, state.tryGet<RespondWithCalled>()) {
+      visitor.visit(respondWithCalled.promise);
+    }
```

Comment:
Does this fix a memory leak?

Reply:
It might. If the `promise` is left dangling and any closures attached to it happen to create a cycle then yes, it could have leaked. A big part of the reason I'm adding this memorytracker mechanism is to help find anything potentially like this.